scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/CloneRepositoryImpl.java

```
@@ -0,0 +1,129 @@
+import edu.asu.diging.simpleusers.core.model.IUser;
+
+@Transactional
+@Service
+public class CloneRepositoryImpl implements CloneRepository{
+
+    @Autowired 
+    private ClonedRepository clonedRepository;
+    
+    private String host;
+    private String owner;
+    private String repo;
+    private String requester;
+    private String url;
+    private ZonedDateTime creationDate;
+    private String drive = "C:/";
+    
+    @Override
+    public ArrayList<Repository> getRepos(){
+        Iterable<RepositoryImpl> repoModels = clonedRepository.findAll();
+        ArrayList<Repository> reposList = new ArrayList<>();
+        repoModels.iterator().forEachRemaining(r -> reposList.add(r));
+        return reposList;
+    }
+
+    @Override
+    public String cloneRepo(CloneForm cloneForm){    
+        
+        host = cloneForm.getHost();
+        owner = cloneForm.getOwner();
+        repo = cloneForm.getRepo();
+        requester = ((IUser)SecurityContextHolder.getContext()
+                            .getAuthentication().getPrincipal()).getUsername();
+        creationDate = ZonedDateTime.now();                                   
+        url = host + "/" + owner + "/" +repo;
+                                           
+        RepositoryImpl repository = new RepositoryImpl();
+        repository.setUrl(url);
+        repository.setHost(host);
+        repository.setOwner(owner);
+        repository.setRepo(repo);
+        repository.setRequester(requester);
+        repository.setCreationDate(creationDate);
+        
+        try{
+            Repository repositoryEntity = clonedRepository.findByUrl(url);
+            if(repositoryEntity != null){
+                clonedRepository.deleteById(repositoryEntity.getId());
+                File parentDirectory = new File(drive + requester + "_" + owner + "_" + repo);
+                deleteDirectoryContents(parentDirectory);
+            }
+        }
+        catch(InvalidDataAccessResourceUsageException e){}
```

do not swallow exceptions (we'll talk about it today actually).